### ISSUE TITLE
figure out way to generate doc from cli.py

### ISSUE THREAD

**Issue by geier** [geier/khal#154]

**Comment by untitaker** [comment]
Without the ``_get_cli`` wrapper in ``cli.py``, you could use autodoc to just document the command functions, I think.

However, I rather suggest to avoid copying the information, and instead fill that page with usage examples.

**Comment by geier** [comment]
and than tell users to just run `khal --help` and `khal command --help`? I find that rather unsatisfying and would much prefer to have the usage documentation in the man page and on the web as well.

**Comment by hobarrera** [comment]
I'm not sure if we should do this ourselves, or actually have click somehow generate it (I'm unfamiliar with the click codebase, but it seems like the place to do it).

I don't like the idea of *copying* stuff, but a web page with the documentation (as well as man docs) would be great. Some CI thing somewhere can probably build/publish the web version if we manage to get this working programatically.

**Comment by untitaker** [comment]
Pretty sure it's somehow possible to just include this into the rtd build process.

The easiest thing would be to use CliRunner to get the --help pages from khal.

**Comment by dvzrv** [comment]
Just to make sure, I get this right: Generating the doc files/man pages is still somewhat "in the works", right?
Seems the current dev version maintainer for Arch Linux thought this is already very much functioning: https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=khal-git
Problem is, that this doesn't build, if you don't have khal installed already.
The doc/config.py makes it fail due to the khal import line.

**Comment by geier** [comment]
No, generating doc files/man pages works. Most parts are static and in `.rst` files in `/doc/` and some other parts get generated. There are only some help texts, those printed when calling `khal command --help` that currently are manually copied over to the `/doc/` folder.

**Comment by geier** [comment]
see https://sphinx-click.readthedocs.io/en/latest/usage/

**Comment by hobarrera** [comment]
For todoman, I use [`sphinx-autorun`](https://github.com/hobarrera/sphinx-autorun). So including this sample below in the `rst` files results in the output of `todo --help` being injected into the final doc. `sphinx-click` Is definitely worth a look though.

```rst
.. runblock:: console

    $ todo --help
```
